src/sonic-config-engine/minigraph.py

```diff
@@ -1001,40 +1002,40 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
     # set port description if parsed from deviceinfo
     for port_name in port_descriptions:
         # ignore port not in port_config.ini
-        if not ports.has_key(port_name):
+        if port_name not in ports:
             continue
 
         ports.setdefault(port_name, {})['description'] = port_descriptions[port_name]
 
-    for port_name, port in ports.items():
+    for port_name, port in list(ports.items()):
         if not port.get('description'):
-            if neighbors.has_key(port_name):
+            if port_name in neighbors:
                 # for the ports w/o description set it to neighbor name:port
                 port['description'] = "%s:%s" % (neighbors[port_name]['name'], neighbors[port_name]['port'])
             else:
                 # for the ports w/o neighbor info, set it to port alias
                 port['description'] = port.get('alias', port_name)
 
     # set default port MTU as 9100
-    for port in ports.itervalues():
+    for port in ports.values():
         port['mtu'] = '9100'
 
     # asymmetric PFC is disabled by default
-    for port in ports.itervalues():
+    for port in ports.values():
         port['pfc_asym'] = 'off'
 
     # set physical port default admin status up
     for port in phyport_intfs:
         if port[0] in ports:
             ports.get(port[0])['admin_status'] = 'up'
 
-    for member in pc_members.keys() + vlan_members.keys():
+    for member in list(pc_members.keys()) + list(vlan_members.keys()):
         port = ports.get(member[1])
         if port:
             port['admin_status'] = 'up'
 
-    for port in neighbors.keys():
-        if port in ports.keys():
+    for port in list(neighbors.keys()):
```

Comment:
```suggestion
    for port in neighbors:
```

Reply:
I see that 2to3 is sticking with the semantic of python 2. And so, since .keys() is not longer a list, it does convert it into a list.